libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -212,20 +218,58 @@ fn generate_docs(
 ) -> Result<PackageDoc, Box<dyn std::error::Error>> {
     // construct the package documentation
     // use type inference to determine types of all values
-    let mut doc = String::new();
+    let mut all_comment = String::new();
     let members = generate_values(&file, &types)?;
     if let Some(comment) = &file.package {
-        doc = comments_to_string(&comment.base.comments);
+        all_comment = comments_to_string(&comment.base.comments);
     }
+    let (headline, description) = seperate_description(&all_comment);
+
     //TODO check if package name exists and if it doesn't throw an error message
     Ok(PackageDoc {
         name: file.package.clone().unwrap().name.name,
-        headline: doc,
-        description: None,
+        headline,
+        description,
         members,
     })
 }
 
+// Separates headline from description
+fn seperate_description(all_comment: &str) -> (String, Option<String>) {
+    let mut headline: String = "".to_string();
+    let mut description = None;
+    let mut reached_end: bool = false;
+    let mut description_text: String = "".to_string();
```

Comment:
I think that these mutables are kind of the bare minimum we would need to split headline and description. I don't think any other implementations of the parser would allow us to eliminate any.

Reply:
I managed to skip over the use of the mut variable `description` entirely to reduce the amount of variables to three.